number_theory/prime.h

```diff
@@ -0,0 +1,27 @@
+#ifndef NUMBER_THEORY_PRIME_H_
+#define NUMBER_THEORY_PRIME_H_
+
+#include <limits>
+
+// This file contains functions related to prime numbers.
+
+namespace number_theory {
+
+// Returns true if |number| is prime.
+template <class T>
+bool is_prime(T number) {
+  static_assert(std::numeric_limits<T>::is_integer,
+                "is_prime argument must be an integer.");
+  // There are no prime numbers smaller than 2.
+  if (number < 2)
+    return false;
+  for (T i = 2; i * i <= number; ++i) {
```

Comment:
Integer overflow may happen during multiplication.

PoC: The following code will make an infinite loop.
```c++
is_prime((uint32_t)4294967291)
```

Reply:
I have switched to `sqrt`. This precision shouldn't be a problem since we decided to enable this template only for small numbers.